apollo-http-cache/src/main/kotlin/com/apollographql/apollo3/cache/http/CachingHttpInterceptor.kt

```diff
@@ -48,17 +49,31 @@ class CachingHttpInterceptor(
         return networkMightThrow(request, chain, cacheKey)
       }
       NETWORK_FIRST -> {
+        var networkException: ApolloException? = null
         try {
           val response = networkMightThrow(request, chain, cacheKey)
           if (response.statusCode in 200..299) {
             //  let HTTP errors through
             return response
           }
         } catch (e: ApolloException) {
-
+          // Original cause of network request failure
+          networkException = e
         }
 
-        return cacheMightThrow(request, cacheKey)
+        try {
+          return cacheMightThrow(request, cacheKey)
+        } catch (cacheMissException: HttpCacheMissException) {
+          // In case of response status code not in 200..299 throw cache miss exception
+          if (networkException == null) throw cacheMissException
```

Comment:
With the suggestion above, this is not needed

Reply:
Agree